packages/headless/src/features/redirection/redirection-slice.ts

```diff
@@ -1,11 +1,26 @@
 import {createReducer} from '@reduxjs/toolkit';
+import {executeSearch} from '../search/search-actions';
 import {checkForRedirection} from './redirection-actions';
 import {getRedirectionInitialState} from './redirection-state';
+import {
+  Trigger,
+  TriggerRedirect,
+  isTriggerRedirect,
+} from './../../api/search/trigger';
 
 export const redirectionReducer = createReducer(
   getRedirectionInitialState(),
   (builder) =>
-    builder.addCase(checkForRedirection.fulfilled, (state, action) => {
-      state.redirectTo = action.payload;
-    })
+    builder
+      .addCase(checkForRedirection.fulfilled, (state, action) => {
+        state.redirectTo = action.payload;
+      })
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        const redirectTriggers: Trigger[] = action.payload.response.triggers.filter(
```

Comment:
I thought I had to specify the type, `any` does seem to work. `isTriggerRedirect` only accepts params of the type `Trigger` so I don't see how I would be able to filter for only `TriggerRedirect` objects.

Reply:
Should the triggers feature live in it's own slice?

It seems like a single state property now gets it's value from two different api calls (`/plan` and `/search`). I think we should separate the two concerns,

Reply:
Are you thinking of doing one slice for all the triggers or having one slice per type of trigger?

Reply:
I was thinking one slice for all triggers, however it would be a good to spend some time thinking about the design, comparing pros and cons if different approaches.

Reply:
I think we should create a separate slice for triggers.

As is, if someone is using a standalone search box and a trigger, if the plan endpoint returns a url, then headless would log a redirection trigger event to UA which would be incorrect.

Reply:
Goind point, agreed, the said slice's state should be split into the triggers types.

A con: would the redirection-slice become misleading then, if it's not affected by trigger on the /search call, but only on the /plan call ?